Content.Server/GameObjects/EntitySystems/LifeSystem.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Text;
+using Content.Shared.GameObjects.EntitySystemMessages;
+using Content.Shared.Input;
+using Content.Server.GameObjects.Components.Mobs.Body;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Server.Interfaces.Player;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.Input;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.GameObjects.Components;
+using Robust.Shared.Interfaces.Timing;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+using Robust.Shared.Players;
+using Robust.Shared.Utility;
+
+
+namespace Content.Server.GameObjects.EntitySystems
+{
+    public interface IOnLife
+    {
+        void OnUpdate(float frameTime);
+    }
+
+    public class LifeSystem : EntitySystem
```

Comment:
I don't think this entity system should be a "generic" system for life components. It would be better if it were specific to `BodyComponent` IMO.

Reply:
i need to use the Update function

Reply:
I meant, make something like a `BodySystem` instead and throw out this `IOnLife` component.